pkg/channel/consolidated/dispatcher/consumer_message_handler.go

```diff
@@ -69,16 +74,33 @@ func (c consumerMessageHandler) Handle(ctx context.Context, consumerMessage *sar
 	ctx, span := tracing.StartTraceFromMessage(c.logger, ctx, message, consumerMessage.Topic)
 	defer span.End()
 
-	_, err := c.dispatcher.DispatchMessageWithRetries(
+	te := kncloudevents.TypeExtractorTransformer("")
+
+	bufferedMessage, err := buffering.CopyMessage(ctx, message, &te)
```

Comment:
Why are you doing this? You also don't free that memory invoking `Message.Finish`, so this is going to leak memory

Reply:
I just followed what we did in the past https://github.com/knative/eventing/blob/main/pkg/channel/fanout/fanout_message_handler.go#L177

Reply:
My assumption is bufferedMessage is managed internally.

Reply:
nope, please check the godocs for that method, it explains how it works

Reply:
Have a pointer? We discussed this in the past and this is the API you mentioned to use.

Reply:
@slinkydeveloper @matzew ?

Reply:
Ok I think you're right, this seems correct, sorry for blocking this. Although I would love to explore if there's any way we can avoid this buffering...

Reply:
@slinkydeveloper np, yeah copying is not something I want to do either, especially on this path, can we do something more efficient at the framework side?

Reply:
yeah maybe you can just pass that transformer to executeRequest then?

Reply:
Let me try it and see if it works.

Reply:
It does work as expected so I will do the update upstream and also here.